### PR TITLE
Add support for mixed forms using libCEED.

### PR DESCRIPTION
**Pull request by YohannDudouit**
This PR generalizes the integration of libCEED to support mixed forms.

 | PR | Author | Editor | Reviewers | Assignment | Approval | Merge|
 | --- | --- | --- | --- | --- | --- | --- |
| [#2912](https://github.com/mfem/mfem/pull/2912) | @YohannDudouit | @tzanio | @acfisher + @tzanio | 4/5/22 | 4/9/22 | 4/14/22 | |

**PR Checklist**
    
- [x] Code builds.
- [x] Code passes `make style`.
- [ ] Update `CHANGELOG`?
- [ ] Update `INSTALL`?
- [ ] All new public, protected, and private classes, methods, data members, and functions have full Doxygen-style documentation in source comments. Documentation should include descriptions of member data, function arguments and return values, template parameters, and prerequisites for calling new functions.
- [ ] Pointer arguments and return values must specify whether ownership is being transferred or lent with the call.
- [ ] Any new functions should include descriptions of their intended use e.g. for internal use only, user-facing, etc., along with references to example code whenever possible/appropriate.
- [ ] Consider adding new sample runs in existing examples to highlight the new capability.

### CODE REVIEW

**Comment by tzanio** [comment]
This PR is now [under review](https://github.com/mfem/mfem/blob/master/CONTRIBUTING.md#pull-requests) (see the table in the PR description). To help with the review process, please **do not force push** to the branch.

**Comment by tzanio** [comment]
Do we have an example that is using this?

Is it worth mentioning in `CHANGELOG`?

**Comment by YohannDudouit** [comment]
> Do we have an example that is using this?
> 
> Is it worth mentioning in `CHANGELOG`?

It's an improvement which is part of a collection of enhancement coming for the libCEED integration. Currently mixed forms can only mix H1 and L2 spaces, however this will become more useful when H(curl) and H(div) support will be added too.

File: fem/ceed/integrator.hpp [hunk 1]
```diff
@@ -119,17 +144,38 @@ class PAIntegrator : public ceed::Operator
...
                     };
       CeedInt nqpts, nelem = mesh.GetNE();
-      CeedInt dim = mesh.SpaceDimension(), vdim = fes.GetVDim();
+      CeedInt dim = mesh.SpaceDimension();
+      CeedInt trial_vdim = trial_fes.GetVDim();
+      CeedInt test_vdim = test_fes.GetVDim();
 
       mesh.EnsureNodes();
-      InitBasisAndRestriction(fes, irm, ceed, &basis, &restr);
+      if ( &trial_fes == &test_fes )
```
Comment (acfisher) [inline fem/ceed/integrator.hpp:152]: I guess the FES object doesn't have an operator== overload defined.

**Review by acfisher — APPROVED** [review]

File: fem/ceed/integrator.hpp (hunk 1 above)
Comment (YohannDudouit) [reply fem/ceed/integrator.hpp:152]: I think using pointer comparison in this context is okay, but I agree that using `==` would be safer, cleaner, and more generic.

**Review by tzanio — APPROVED** [review]

**Comment by tzanio** [comment]
Merged in `next` for testing...

**Pull request merged**